pysat/instruments/nasa_cdaweb_methods.py

```diff
@@ -260,27 +262,61 @@ def download(supported_tags, date_array, tag, sat_id,
         saved_local_fname = os.path.join(data_path, formatted_local_fname)
 
         # perform download
-        try:
-            print('Attempting to download file for '+date.strftime('%x'))
-            sys.stdout.flush()
-            remote_path = '/'.join((remote_url, formatted_remote_fname))
-            req = requests.get(remote_path)
-            if req.status_code != 404:
-                open(saved_local_fname, 'wb').write(req.content)
-                print('Finished.')
-            else:
-                print('File not available for ' + date.strftime('%x'))
-        except requests.exceptions.RequestException as exception:
-            print('File not available for ' + date.strftime('%x'))
+        if not multi_file_day:
+            try:
+                print('Attempting to download file for ' + date.strftime('%x'))
+                sys.stdout.flush()
+                remote_path = '/'.join((remote_url.strip('/'),
+                                        formatted_remote_fname))
+                req = requests.get(remote_path)
+                if req.status_code != 404:
+                    open(saved_local_fname, 'wb').write(req.content)
+                    print('Finished.')
+                else:
+                    print('File not available for ' + date.strftime('%x'))
+            except requests.exceptions.RequestException as exception:
+                print(' '.join((exception, '- File not available for',
+                                date.strftime('%x'))))
+
+        else:
+            try:
+                print('Attempting to download files for '
+                      + date.strftime('%x'))
+                sys.stdout.flush()
+                remote_files = list_remote_files(tag=tag, sat_id=sat_id,
+                                                 supported_tags=supported_tags,
+                                                 year=date.year,
+                                                 month=date.month,
+                                                 day=date.day)
```

Comment:
Won't date always have a year, month, and day here? I'm a bit confused as to how the list_remote_files code for month or day is None is run in practice.

Reply:
The download routine assumes you are downloading one or more files per day.  When download calls list_remote_files, it fills in all these values.  If the user wants to just look for data in a given year, they can specify only the year, and month and day will be None automatically.

Right now, internal stuff assumes the exact date exists, but the code is flexible enough to allow the user to search for larger ranges.

Reply:
I think it's clearer to think of the `list_remote_files` as a method that can either be called by the user 'directly' (through remote_list_files) or indirectly through the `download` method.

Examples:
```
d0 = pysat.datetime(2009, 1, 1)
saber.download(d0, d0)
```
will download all data for 1 Jan 2009.  `download` determines which files need to be downloaded by passing through all three variables (as in the `date` statements above) to `list_remote_files`.   Since the `download` method assumes one day at a time, it always sets all three date parameters for this call.

Conversely,
```
saber.remote_file_list(year=2009, month=1)
```
is a user-invoked command that will call `list_remote_files`, which fills in missing values as None.  In this case, day=None, and all files for January are returned.

While writing this up, I realized that I probably should set up proper variables in the `remote_files_list` method instead of just passing everything through as `**kwargs` and letting cdaweb_methods sort it out.

Reply:
OK, thanks for clearing that up.